src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/ConvertDataControllerTests.cs

```diff
@@ -147,7 +197,72 @@ private static Parameters GetParamsResourceWithTooManyParams()
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.TemplateCollectionReference, Value = new FhirString("microsofthealth/fhirconverter:default") });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.RootTemplate, Value = new FhirString("ADT_A01") });
+
+            return parametersResource;
+        }
+
+        private static Parameters GetParamsResourceWithInconsistentParamsWrongDefaultTemplates()
+        {
+            var parametersResource = new Parameters();
+            parametersResource.Parameter = new List<Parameters.ParameterComponent>();
+
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputData, Value = new FhirString(GetSampleCcdaMessage()) });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputDataType, Value = new FhirString("Ccda") });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.TemplateCollectionReference, Value = new FhirString("microsofthealth/fhirconverter:default") });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.RootTemplate, Value = new FhirString("CCD") });
+
+            return parametersResource;
+        }
+
+        private static Parameters GetParamsResourceWithInconsistentParamsWrongHl7v2DefaultTemplates()
+        {
+            var parametersResource = new Parameters();
+            parametersResource.Parameter = new List<Parameters.ParameterComponent>();
+
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputData, Value = new FhirString(GetSampleCcdaMessage()) });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputDataType, Value = new FhirString("Ccda") });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.TemplateCollectionReference, Value = new FhirString("microsofthealth/hl7v2templates:default") });
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.RootTemplate, Value = new FhirString("CCD") });
+
+            return parametersResource;
+        }
+
+        private static Parameters GetParamsResourceWithInconsistentParamsWrongCcdaDefaultTemplates()
+        {
+            var parametersResource = new Parameters();
+            parametersResource.Parameter = new List<Parameters.ParameterComponent>();
+
+            parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputData, Value = new FhirString(GetSampleHl7v2Message()) });
```

Comment:
>parametersResource.Parameter.Add(new Parameters.ParameterComponent() { Name = ConvertDataProperties.InputData, Value = new FhirString(GetSampleHl7v2Message()) }); [](start = 12, length = 162)

Considering how often this pattern repeats, I would suggest to add 
```
private static void AddParamComponent(Parameters resource, string name, string value)=>
resource.Parameter.Add(new Parameters.ParameterComponent() { Name =name, Value =new FhirString(value)});
```
and just do:
`AddParamComponent(parametersResource,ConvertDataProperties.InputDataType, "Hl7V2");`

Reply:
Good idea. I will do it.